vmm/src/lib.rs

```diff
@@ -1143,6 +1148,23 @@ impl Vmm {
         Ok(())
     }
 
+    #[cfg(target_arch = "x86_64")]
+    fn get_serial_device(&self) -> Option<Arc<Mutex<RawIOHandler>>> {
+        Some(self.legacy_device_manager.stdio_serial.clone())
+    }
+
+    #[cfg(target_arch = "aarch64")]
+    fn get_serial_device(&self) -> Option<Arc<&Mutex<RawIOHandler>>> {
+        let raw_io_device = self
```

Comment:
This function can be rewritten to:

```
    #[cfg(target_arch = "aarch64")]
    fn get_serial_device(&self) -> Option<&Arc<Mutex<RawIOHandler>>> {
        self
            .mmio_device_manager
            .as_ref()
            .unwrap()
            .get_raw_io_device(DeviceType::Serial, "uart")
    }
```
if you change its return result inside the `mmiodevicemanager`.